src/app/stop-watch/stop-watch.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from '@angular/core';
+import { interval, Subject, takeUntil, debounceTime, buffer, filter, map } from "rxjs";
+
+@Component({
+  selector: 'app-stop-watch',
+  templateUrl: './stop-watch.component.html',
+  styleUrls: ['./stop-watch.component.scss']
+})
+export class StopWatchComponent implements OnInit {
+  time: number = 0;
+
+  private observer = new Subject();
+  private rxjsTimer = interval(1000);
+
+  private delay: number = 0;
+  private subscription: any;
+
+  obs = new Subject();
+
+  startTimer() {
+    this.subscription = this.rxjsTimer.pipe(takeUntil(this.observer))
+      .subscribe(
+        (counter) => {
+          this.time = counter + this.delay;
+        })
+  }
+  stopTimer() {
+    this.stopIteartion()
+    this.time = 0;
+    this.delay = 0;
+
+  }
+
+  stopIteartion() {
+    this.observer.next('');
+    this.observer.complete();
+    this.observer = new Subject();
+  }
+
+  public ngOnInit() {
```

Comment:
we have following structure for class:
public variables
private variables
constructor
lifecycle methods
public custom methods
private custom methods

please move accordingly